R/helpers-where.R

```diff
@@ -67,7 +67,11 @@ where <- function(fn) {
     out <- predicate(x, ...)
 
     if (!is_bool(out)) {
-      abort("`where()` must be used with functions that return `TRUE` or `FALSE`.")
+      not <- obj_type_friendly(out)
+      abort(
+        glue("Predicate must return a single `TRUE` or `FALSE`, not {not}."),
+        call = quote(where())
```

Comment:
I think this is correct because `where()` is no longer on the stack, and while this is technically lying about where the error comes from, it's more informative for the user.